vendor.mod

```diff
@@ -15,7 +15,7 @@ require (
 	github.com/Microsoft/go-winio v0.5.2
 	github.com/Microsoft/hcsshim v0.9.5
 	github.com/RackSec/srslog v0.0.0-20180709174129-a4725f04ec91
-	github.com/armon/go-radix v0.0.0-20180808171621-7fddfc383310
+	github.com/armon/go-radix v1.0.1-0.20221118154546-54df44f2176c
```

Comment:
In case you change your mind and prefer a better maintained repo: https://github.com/moby/moby/compare/master...tiborvass:immutable_radix?expand=1

Reply:
That was my initial attempt in https://github.com/moby/libnetwork/pull/2581, but (don't recall exactly) I think it didn't work. But must admit that I didn't dive deeply into why (or forgot that I did) 😂

Reply:
What was the problem? From looking at your branch, what possibly didn't work is that you take the lock twice. createOrUpdateEntry should not have a lock because the caller has to lock.

I'm opening https://github.com/moby/moby/pull/44501 we can discuss it there.